text_collector_examples/ipmitool

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/awk -nf
+
+#
+# Converts output of `ipmitool sensor` to prometheus format.
+#
+# With GNU awk:
+#   ipmitool sensor | ./ipmitool > ipmitool.prom
+#
+# With BSD awk:
+#   ipmitool sensor | awk -f ./ipmitool > ipmitool.prom
+#
+
+function discrete2float(hex_value) {
+	return hex_value + 0;
+}
+
+function exportArray(a, name) {
+	if (a["metric_count"] < 1) {
+		return
+	}
+	delete a["metric_count"]
+
+	if (name != "status") {
+		printf("# TYPE %s%s gauge\n", namespace, name);
```

Comment:
This also needs to print a `# HELP metric_name Something`.

Reply:
Happy to add something.  Was going to ask about this...  Is a comment like , "Readings from the {voltage,temperature,etc} sensors via ipmitool, values are machine-dependent" valuable?  (What I would add, if left to my own devices.)   I don't think I can interpret very well beyond this.  Is it desirable to have a comment that adds no additional information?

Additionally, I'd like to mention about the shebang...  I've defaulted it to work in GAWK, because that's the most likely target, but the shebang syntax doesn't work on BSD.  It's common to patch this in the ports tree though, and alternate syntax is provided.

Reply:
`Readings from the {voltage,temperature,etc} sensors via ipmitool` should be sufficient.  We don't really do anything with this data, yet, but having it there is useful for future-proofing.

I think the shebang is fine.

Reply:
What about skipping the type on status?  I feel that because there is no linear relationship between possible values, it isn't really a gauge (nor any of the other possible types, of course).  Is it simply that the value goes both up and down that makes it a gauge?